app/controllers/drafts/ViewAllDraftMovementsController.scala

```diff
@@ -0,0 +1,115 @@
+import models.response.{ErrorResponse, NotFoundError}
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import viewmodels.draftMovements.DraftMovementsPaginationHelper
+import viewmodels.helpers.SelectItemHelper
+import views.html.viewAllDrafts.ViewAllDraftMovementsView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ViewAllDraftMovementsController @Inject()(mcc: MessagesControllerComponents,
+                                                getDraftMovementsConnector: GetDraftMovementsConnector,
+                                                view: ViewAllDraftMovementsView,
+                                                getExciseProductCodesConnector: GetExciseProductCodesConnector,
+                                                errorHandler: ErrorHandler,
+                                                val auth: AuthAction,
+                                                val getData: DataRetrievalAction,
+                                                val betaAllowList: BetaAllowListAction,
+                                                formProvider: ViewAllDraftMovementsFormProvider,
+                                                paginationHelper: DraftMovementsPaginationHelper
+                                          )(implicit val executionContext: ExecutionContext, appConfig: AppConfig)
+  extends FrontendController(mcc)
+    with AuthActionHelper with I18nSupport with BetaChecks {
+
+  def onPageLoad(ern: String, searchOptions: GetDraftMovementsSearchOptions): Action[AnyContent] = {
+    authorisedDataRequestAsync(ern, searchMovementsBetaGuard(ern)) { implicit request =>
+      renderView(Ok, ern, searchOptions, formProvider())
+    }
+  }
+
+  def onSubmit(ern: String, searchOptions: GetDraftMovementsSearchOptions): Action[AnyContent] = {
+    authorisedDataRequestAsync(ern, searchMovementsBetaGuard(ern)) { implicit request =>
+      formProvider().bindFromRequest().fold(
+        formWithErrors => renderView(BadRequest, ern, searchOptions, formWithErrors),
+        value => Future(Redirect(routes.ViewAllDraftMovementsController.onPageLoad(ern, value)))
+      )
+    }
+  }
+
+  private def renderView(
+                          status: Status,
+                          ern: String,
+                          searchOptions: GetDraftMovementsSearchOptions,
+                          form: Form[GetDraftMovementsSearchOptions]
+                        )(implicit request: DataRequest[_]): Future[Result] = {
+
+    val result: EitherT[Future, ErrorResponse, Result] = for {
+      draftMovements <- EitherT(getDraftMovementsConnector.getDraftMovements(ern, Some(searchOptions)).map {
+        case Left(NotFoundError) => Right(GetDraftMovementsResponse(0, Seq.empty))
```

Comment:
I don't think this should happen? I think the connector would never return a `Left(NotFoundError)` because the backend always returns an `OK` on success with a JSON body. That body would already contain `0` for the count and an empty `Seq()`